flang/lib/Parser/prescan.cpp

```diff
@@ -997,14 +997,12 @@ const char *Prescanner::FixedFormContinuationLine(bool mightNeedSpace) {
   }
   tabInCurrentLine_ = false;
   char col1{*nextLine_};
-  if (InCompilerDirective()) {
```

Comment:
By deleting the contextual test `if (InCompilerDirective)`, this code will apply to lines that do not follow an initial compiler directive line.  This will allow continuation to take place if an apparent compiler directive continuation line follows a source line that is not a compiler directive; it will also allow line continuation when an apparent compiler directive continuation line's sentinel does not match the sentinel of the original compiler directive (e.g., `!$OMP&` after an `!$ACC` line).  Is this new behavior intentional?

Reply:
Like luporl said, the following code should be legal, even if there was no sentinel in the initial line.
```fortran
      k01=1+
!$   &1
```
And the new commit reinstates the check for consistency between the continuation line's sentinel and the initial line's.